Developer setup documentation update (#498)
SwissDataScienceCenter/renku: docs/developer/python_setup.rst

.. _python_setup:

Python setup
============

**Renku** requires ``python`` 3.6+.

Installation (MacOS users)
--------------------------

This setup assumes you've got ``Homebrew`` installed in your computer. If not please look `here <https://brew.sh>`_.
To check which version of python you've got already installed you can use:

.. code-block:: console

    $ python -version

If you have python 2 you can consider uninstalling it with

.. code-block:: console

    $ brew uninstall --force python@2

> Before installing Python 3 add a following export line to ``~/.profile``

Reviewer: ```suggestion
Before installing Python 3 add the following line to ``~/.profile``
```

.. code-block:: console

>     $ export PATH="/usr/local/opt/python/libexec/bin:$PATH"

Reviewer: I don't have this in my `PATH` -- why is it needed?

Author: I'm not entirely sure but I think this is to override the default osx' python so the brew installed python is kind of hiding the system one.

Reviewer 2: IMHO it's not necessary. If there is a problem with Python binary installed by brew, one should just run `brew doctor`.

Reviewer: I think @jirikuncar's point is that brew should take care of making sure that the brew-installed versions are the ones that are actually used. But you might be right that it doesn't always work out that way...

Author: My experience is that when it comes into binaries both provided by OSX natively and installed with `brew`, the default is that the system ones are winning as they are on the `PATH` and the additional ones not necessarily. And the problem is that for `brew` everything is fine as from its perfectly and installation wise all is good. But maybe I'm doing something wrong.
